### ISSUE TITLE
0.17.0-beta.16: System.define breaking System.import

### ISSUE THREAD

**Issue by m-a-r-c-e-l-i-n-o** [jspm/jspm-cli#1863]
Not sure if this is a jspm issue, systemjs issue, or some other configuration issue but `System.import` is having trouble importing "esm/es6" modules defined through `System.define`. Here's what I tried:
```
System.define(System.normalizeSync('testing.js'), 'var hello = "world"')
System
.import(System.normalizeSync('testing.js'))
.then(function (module) {
    console.log(module)
})
```
This works just fine. But doing the same, except with:
```
System.define(
    System.normalizeSync('testing.js'),
    'export default function () { console.log('Testing FAKE module!') }'
) 
```
Throws: 
```
TypeError: Cannot read property 'split' of undefined
Error loading /home/marcbraulio/Projects/jspm-mock/test/testing.js
```
So I tried:
```
System.define(
    System.normalizeSync('testing.js'),
    'export default function () { console.log('Testing FAKE module!') }',
    { metadata: { format: 'global' } }
) 
```
And got:
```
SyntaxError: Unexpected token export
	    at Object.exports.runInThisContext (vm.js:53:16)
	Evaluating undefined
	Error loading /home/marcbraulio/Projects/jspm-mock/test/testing.js
```
I think this indicates that the issue is with the "esm/es6" formats, I just haven't been able to pin it down just yet. No issues importing "esm/es6" modules directly without running them through `System.define`. I also had no problems with the other formats. I did some brief tests with jspm 0.16, and the issue does not seem to be present there either. Thank you for your time!

**Comment by guybedford** [comment]
Sure, I've added a fix in https://github.com/ModuleLoader/es6-module-loader/commit/09f19775e16a03ad516eeeb3d689c232d651adce.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.18.
